### PR TITLE
fix(rpc-fns): Read state gas limit restrictions

### PR DESCRIPTION
**Pull request by breathx**

### CODE REVIEW

File: pallets/gear/src/runtime_api.rs [hunk 1]
```diff
@@ -302,7 +302,7 @@ where
             None,
             None,
             payload,
-            BlockGasLimitOf::<T>::get() / 4,
+            BlockGasLimitOf::<T>::get() * 6,
```
Comment (shamilsan) [inline pallets/gear/src/runtime_api.rs:305]: Isn't it a hack? Looks like magic.

**Review by shamilsan — APPROVED** [review]

**Review by ark0f — APPROVED** [review]

File: pallets/gear/src/runtime_api.rs (hunk 1 above)
Comment (breathx) [reply pallets/gear/src/runtime_api.rs:305]: Not exactly. Just previous limitations became under required

File: core-processor/src/executor.rs [hunk 2]
```diff
@@ -602,6 +602,8 @@ where
         .into_ext_info(&memory)
         .map_err(|e| format!("Backend postprocessing error: {e:?}"))?;
 
+    log::debug!("\n\n\tBURNED: {}\n\n", info.gas_amount.burned());
```
Comment (grishasobol) [inline core-processor/src/executor.rs:605]: remove this debug or change to trace and remove this garbage  \n

File: pallets/gear/src/runtime_api.rs (hunk 1 above)
Comment (grishasobol) [inline pallets/gear/src/runtime_api.rs:305]: If your are using * 6 for each, then this must be separate constant with explanation why * 6.

File: pallets/gear/src/tests.rs [hunk 3]
```diff
@@ -77,12 +74,57 @@ use gear_core_errors::*;
...
+                payload.encode(),
+                BlockGasLimitOf::<Test>::get(),
+                0
+            ));
+            let mid = get_last_message_id();
+
+            run_to_next_block(None);
+
+            assert_succeed(mid);
+            let state = Gear::read_state_impl(pid).expect("Failed to read state");
```
Comment (grishasobol) [inline pallets/gear/src/tests.rs:117]: `state` is not using in test. At least the size must be assert_eq!

**Pull request merged**